tests/test_time_utils.py

```diff
@@ -0,0 +1,125 @@
+        """
+        # do tests at random points between epoch and max. time
+        times = [
+            "1970-01-01T12:00:00.123",
+            "1999-12-31T23:59:57.123456",
+            "2000-01-01T12:00:00.123456",
+            "2030-01-01T12:00:00.123456789",
+            "2075-08-17T00:03:45",
+            "2099-12-31T23:00:50",
+        ]
+        for time in times:
+            atime = Time(time, format="isot", scale="tai")
+            for sec in range(7):
+                # loop over few seconds to add to each time
+                for i in range(100):
+                    # loop over additional fractions of seconds
+                    delta = sec + 0.3e-9 * i
+                    in_time = atime + TimeDelta(delta, format="sec")
+                    # do round-trip conversion to nsec and back
+                    value = time_utils.astropy_to_nsec(in_time)
+                    value = time_utils.nsec_to_astropy(value)
+                    delta2 = value - in_time
+                    delta2_sec = delta2.to_value("sec")
+                    # absolute precision should be better than half
+                    # nanosecond, but there are rounding errors too
+                    self.assertLess(abs(delta2_sec), 0.51e-9)
+
+    def test_times_equal(self):
+        """Test for times_equal method
+        """
+        # time == time should always work
+        time1 = Time("2000-01-01T00:00:00.123456789", format="isot", scale="tai")
+        self.assertTrue(time_utils.times_equal(time1, time1))
+
+        # one nsec difference
+        time1 = Time("2000-01-01T00:00:00.123456789", format="isot", scale="tai")
+        time2 = Time("2000-01-01T00:00:00.123456788", format="isot", scale="tai")
+        self.assertTrue(time_utils.times_equal(time1, time2, 2.))
+        self.assertTrue(time_utils.times_equal(time2, time1, 2.))
+        self.assertFalse(time_utils.times_equal(time1, time2, .5))
+        self.assertFalse(time_utils.times_equal(time2, time1, .5))
+
+        # one nsec difference, times in UTC
+        time1 = Time("2000-01-01T00:00:00.123456789", format="isot", scale="utc")
+        time2 = Time("2000-01-01T00:00:00.123456788", format="isot", scale="utc")
+        self.assertTrue(time_utils.times_equal(time1, time2, 2.))
+        self.assertTrue(time_utils.times_equal(time2, time1, 2.))
+        self.assertFalse(time_utils.times_equal(time1, time2, .5))
+        self.assertFalse(time_utils.times_equal(time2, time1, .5))
+
+        # 1/2 nsec difference
+        time1 = Time("2000-01-01T00:00:00.123456789", format="isot", scale="tai")
+        time2 = time1 + TimeDelta(0.5e-9, format="sec")
+        self.assertTrue(time_utils.times_equal(time1, time2))
+        self.assertTrue(time_utils.times_equal(time2, time1))
+        self.assertFalse(time_utils.times_equal(time1, time2, .25))
+        self.assertFalse(time_utils.times_equal(time2, time1, .25))
+
+        # 1/2 mksec difference
```

Comment:
typo: mksec

Reply:
Do you prefer μsec? 🙂

Reply:
I've never seen mk used for micro before so I think I'd prefer (microsec) -- but of course a μ should work fine in the source code so I'm also fine with that...